src/test/java/net/openhft/chronicle/queue/bench/QueueSparseFilesJLBHBenchmark.java

```diff
@@ -0,0 +1,109 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.openhft.chronicle.queue.bench;
+
+import net.openhft.chronicle.core.io.IOTools;
+import net.openhft.chronicle.core.util.NanoSampler;
+import net.openhft.chronicle.jlbh.JLBH;
+import net.openhft.chronicle.jlbh.JLBHOptions;
+import net.openhft.chronicle.jlbh.JLBHTask;
+import net.openhft.chronicle.queue.ExcerptAppender;
+import net.openhft.chronicle.queue.ExcerptTailer;
+import net.openhft.chronicle.queue.impl.single.SingleChronicleQueue;
+import net.openhft.chronicle.wire.DocumentContext;
+
+import static net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder.single;
+
+/*
+Chronicle Queue performance is compared when using sparse files and chunking. The results illustrate similar performance for both configurations.
+ */
+public class QueueSparseFilesJLBHBenchmark implements JLBHTask {
+    private static int iterations;
+    private SingleChronicleQueue sourceQueue;
+    private SingleChronicleQueue sinkQueue;
+    private ExcerptTailer tailer;
+    private ExcerptAppender appender;
+    private JLBH jlbh;
+    private NanoSampler writeProbe;
+    private NanoSampler readProbe;
+    private static int round = 1;
+
+    public static void main(String[] args) {
+        int throughput = 100_000;
+        int warmUp = 1_000_000;
+        iterations = 500_000;
+
+        JLBHOptions lth = new JLBHOptions()
+                .warmUpIterations(warmUp)
+                .iterations(iterations)
+                .throughput(throughput)
+                .recordOSJitter(false)
+                .skipFirstRun(true)
+                .runs(5)
+                .jlbhTask(new QueueSparseFilesJLBHBenchmark());
+        new JLBH(lth).start();
+    }
+
+    @Override
+    public void init(JLBH jlbh) {
+        IOTools.deleteDirWithFiles("sparseFile", 5);
+        IOTools.deleteDirWithFiles("chunking", 5);
+
+        if (round == 1) {
+            sourceQueue = single("sparseFile").useSparseFiles(true).sparseCapacity(64L << 30).build();
+            sinkQueue = single("sparseFile").useSparseFiles(true).sparseCapacity(64L << 30).build();
+        } else {
+            sourceQueue = single("chunking").blockSize(64L << 30).build();
+            sinkQueue = single("chunking").blockSize(64L << 30).build();
+        }
+        appender = sourceQueue.acquireAppender();
+        tailer = sinkQueue.createTailer().disableThreadSafetyCheck(true);
+        this.jlbh = jlbh;
+        writeProbe = round == 1 ? jlbh.addProbe("write (sparse file)") : jlbh.addProbe("write (chunking)");
+        readProbe = round == 1 ? jlbh.addProbe("read (sparse file)") : jlbh.addProbe("read (chunking)");
+    }
+
+    @Override
+    public void run(long startTimeNS) {
+
+        try (DocumentContext dc = appender.writingDocument()) {
```

Comment:
for sparse files, probably a more suitable test would be to write a message every so often - eg every 1MB, or every 64MB etc, rather than consecutive messages.
To do this, use eg 
````
dc.wire().bytes().writeInt(pos, value); // eg ...writeInt(64L<<20, 1234) to write 1234 at the 64MB point
````
So eg add a long index which starts at 0 then increments by 1L<<20 every write.

Will need to choose a reasonable step size so can have enough points to sample without going too huge on the file. A sparse file of ~1TB containing a message every 1MB for 1M messages would be ok for example (or every 100k for 10M messages)

Also, add a round-3 test which uses a standard blocksize (eg 256L<<20) to compare with the giant block size

Reply:
Now the message size is 1k bytes but the content is only 11 bytes so I think it resembles sparse files.